theano/tensor/type.py

```diff
@@ -204,8 +204,8 @@ def filter_variable(self, other, allow_convert=True):
         Convert a symbolic Variable into a TensorType, if compatible.
 
         For the moment, only a TensorType or CudaNdarrayType will be
```

Comment:
Not scalar for now:

$ python -c "import theano;s=theano.scalar.float32(); ts=theano.tensor.scalar(); ts.type.filter_variable(s)"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "theano/tensor/type.py", line 234, in filter_variable
    self=self))
TypeError: Cannot convert Type float32 (of Variable <float32>) into Type TensorType(float64, scalar). You can try to manually convert <float32> into a TensorType(float64, scalar).


But maybe we should. Anyway, too close to an rc to change that now. I'll update about GpuArrayType.